### PR TITLE
Pkm ev

### PR DESCRIPTION
**Pull request by Edenrol**
add ev to scraper and view in pokedex

### CODE REVIEW

File: bin/scrape-pokeapi.ts [hunk 1]
```diff
@@ -81,6 +82,7 @@ interface PokemonSimple {
   spDefense: number;
   speed: number;
   id: string;
+  efforts: Record<string, number>;
```
Comment (wavebeem) [inline bin/scrape-pokeapi.ts:85]: Let's use type safety here :)

```ts
efforts: {
  hp: number;
  attack: number;
  spAttack: number;
  defense: number;
  spDefense: number;
  speed: number;
}
```

File: bin/scrape-pokeapi.ts [hunk 2]
```diff
@@ -140,6 +142,11 @@ export async function scrapePokeapi(): Promise<void> {
         key: (item) => item.stat.name,
         value: (item) => item.base_stat,
       });
+      const efforts = toObject({
+        data: detail.stats,
+        key: (item) => item.stat.name,
+        value: (item) => item.effort,
+      });
```
Comment (wavebeem) [inline bin/scrape-pokeapi.ts:149]: Let's change these names like `special-attack` ➡️  `spAttack` to match the other keys

File: bin/scrape-pokeapi.ts [hunk 3]
```diff
@@ -163,10 +170,12 @@ export async function scrapePokeapi(): Promise<void> {
         spDefense: stats["special-defense"] ?? 0,
         speed: stats["speed"] ?? 0,
         id: String(detail.id),
+        efforts: efforts,
         types: detail.types.map((t) => t.type.name),
       };
       pokemonSimpleList.push(mon);
       console.log(speciesDetail.id, detail.id);
+     // console.log(mon);
```
Comment (wavebeem) [inline bin/scrape-pokeapi.ts:178]: You can delete this line now :)

File: src/StatsTable.tsx [hunk 4]
```diff
@@ -26,7 +26,7 @@ export function StatsTable({ pokemon }: StatsTableProps) {
       >
         {t("pokedex.stats.hp")}
       </div>
-      <div className="tr">{hp}</div>
+      <div className="tr">{hp} / {efforts["hp"]}</div>
```
Comment (wavebeem) [inline src/StatsTable.tsx:29]: `efforts["hp"]` ➡️ `efforts.hp`

File: src/StatsTable.tsx [hunk 5]
```diff
@@ -64,7 +64,7 @@ export function StatsTable({ pokemon }: StatsTableProps) {
           ? t("pokedex.stats.specialAttackLong")
           : t("pokedex.stats.specialAttack")}
       </div>
-      <div className="tr">{spAttack}</div>
+      <div className="tr">{spAttack} / {efforts["special-attack"]}</div>
```
Comment (wavebeem) [inline src/StatsTable.tsx:67]: `efforts["special-attack"]` ➡️ `efforts.spAttack`

File: src/data-types.ts [hunk 6]
```diff
@@ -17,6 +17,7 @@ export interface Pokemon {
   spDefense: number;
   speed: number;
   hasShiny: boolean;
+  efforts: Record<string, string>;
```
Comment (wavebeem) [inline src/data-types.ts:20]: Let's use the other type from the top

File: src/StatsTable.tsx [hunk 7]
```diff
@@ -99,11 +99,13 @@ export function StatsTable({ pokemon }: StatsTableProps) {
         aria-label={hasRoomForFullText ? "" : t("pokedex.stats.totalLong")}
       >
         {hasRoomForFullText
-          ? t("pokedex.stats.totalLong")
-          : t("pokedex.stats.total")}
+          ? t("pokedex.stats.totalLong")+ " / EVs"
+          : t("pokedex.stats.total") + " / EVs"}
```
Comment (wavebeem) [inline src/StatsTable.tsx:103]: The term "EVs" would need to be translated... https://bulbapedia.bulbagarden.net/wiki/Effort_values#In_other_languages

**Comment by wavebeem** [comment]
Great work so far! I really like this.

I have a thought, though: what if the EVs were displayed after the graph? I think this is a lot easier to read.

image

**Comment by wavebeem** [comment]
It might be nicer to display a blank space instead of 0 since most EVs are 0, right?

**Comment by Edenrol** [comment]
> Great work so far! I really like this.
> 
> I have a thought, though: what if the EVs were displayed after the graph? I think this is a lot easier to read.
> 
> image

Yes, could be nice to be there.
Olso think that are a missing a reference to the evs, i don't know where the reference could be.

**Comment by wavebeem** [comment]
Yeah... maybe at the top of the page there is a checkbox like:

*✅ Show effort values*

it's a bit tricky though. it needs to fit well even on mobile screens...

*Sage Fennel* (they/them)
wavebeem.com** <https://www.wavebeem.com/>

**Comment by Edenrol** [comment]
The checkbox could be at the configuration page inside the "Settings grid" 

In my opinion is more organised there.

**Comment by wavebeem** [comment]
Sure, that's fine too!

*Sage Fennel* (they/them)
wavebeem.com** <https://www.wavebeem.com/>

**Comment by wavebeem** [comment]
Just tidying up right now. If you still want to work on this, leave a comment and I will re-open the issue.

**Comment by Edenrol** [comment]
Hi, At the moment, I don't have enough knowledge of the technology to
proceed with the topic. I apologize.

**Comment by wavebeem** [comment]
That's ok. If you want to work on this again in the future, I'd be happy to see the contribution! Thanks for your work.

**Pull request closed without merging**